src/main/java/org/dataloader/stats/StatisticsCollector.java

```diff
@@ -11,38 +16,109 @@ public interface StatisticsCollector {
     /**
      * Called to increment the number of loads
      *
+     * @param <K> the class of the key in the data loader
+     * @param context the context containing metadata of the data loader invocation
+     *
      * @return the current value after increment
      */
+    default <K> long incrementLoadCount(IncrementLoadCountStatisticsContext<K> context) {
```

Comment:
In hindsight, rather than parameterise each new method, would it have made more sense to parameterise the entire class on `K`?

Reply:
I explored this in a bit more detail; having to pass in a type parameter into certain contexts (`ThreadLocalStatisticsCollector`'s `static` `collector` and `DataLoaderOptions`) made this awkward; so I'll leave this as-is.